### PR TITLE
[WIP] Add initial cinnamon support to yaru

### PR DESCRIPTION
**Pull request by hsbasu**
Created using [this](https://github.com/linuxmint/mint-themes/tree/master/src/Mint-Y/cinnamon/sass) as base. This is only initial support and many things to be done. For example, proper orange color scheme of *yaru* is not fully implemented yet. Code clean-up and dark-mode needs to be worked on. Hence, it will be changed based on reviews.

**Tasks**
1. [x] Create a copy of all the upstream code (the whole `cinnamon` folder, not just sass) inside a new `cinnamon/upstream/ `folder. This will allow us to keep the theme fully synced with upstream thanks to a GIthub Action.
2. [x] Add `_palette.scss` and `_colors.scss` from `gnome-shell` 
3. [ ] Add/remove extra colour variables
4. [x] Implement basic Orange colour scheme of Yaru
5. [x] Implement dark mode
6. [x] Circle highlights on calendar instead of Mint's rectangle one.
8. [x] Each modifications in the code should be marked with a `// Yaru change: description of the change` comment.

### CODE REVIEW

**Comment by hsbasu** [comment]
@ItzSwirlz and @Jupi007 I have finished the modifications. Please test the theme.

**Review by Jupi007 — CHANGES_REQUESTED** [review]
Thanks a lot @hsbasu 

I have a few notes:

- Can you symlink `default/_drawing.scss` and `default/_common.scss`, to the corresponding upstream file (like that when we'll update upstream files, there no more work needed)?
- The `default/parse-sass.sh` is not needed (same for dark theme). We'll build sass files with Meson. Can you remove it?

File: cinnamon/src/dark/_palette.scss [hunk 1]
```diff
@@ -0,0 +1 @@
+/home/himadri/Projects/Side_projects/my-repos/packages/test/yaru/cinnamon/src/default/_palette.scss
```
[the remark this answers is not in the document]
Comment (hsbasu) [reply cinnamon/src/dark/_palette.scss]: Resolved it.

**Comment by hsbasu** [comment]
> I have a few notes:
> 
>     * Can you symlink `default/_drawing.scss` and `default/_common.scss`, to the corresponding upstream file (like that when we'll update upstream files, there no more work needed)?
https://github.com/ItzSwirlz/yaru/pull/2/commits/9ec3716f39a7a9f526c265061362367ce77f1cb0
> 
>     * The `default/parse-sass.sh` is not needed (same for dark theme). We'll build sass files with Meson. Can you remove it?
https://github.com/ItzSwirlz/yaru/pull/2/commits/8218a5a53aa6326d7b42db30737c0f080bf0d3a7

Done

**Review by Jupi007 — APPROVED** [review]
LGFM :+1:

**Comment by ItzSwirlz** [comment]
Alrighty, thank you everyone!

**Pull request merged**